pkg/sql/catalog/lease/kv_writer.go

```diff
@@ -66,7 +74,17 @@ func (w *kvWriter) versionGuard(
 
 func (w *kvWriter) insertLease(ctx context.Context, txn *kv.Txn, l leaseFields) error {
 	return w.do(ctx, txn, l, func(guard settingswatcher.VersionGuard, b *kv.Batch) error {
-		if guard.IsActive(clusterversion.TODO_Delete_V23_1_SystemRbrDualWrite) {
+		// Write a session based lease, if at the very least we are dual writing
```

Comment:
I'm finding this comment (and the similar one below) very difficult to interpret. I generally expect a comment ahead of an if to be in the format `why, action` as that matches the flow of control. Alternatively, `action, why` can be placed within the if block.

The `isSessionBasedLeasingModeActive` method is also quite confusing as it implies that the mode is an enum but the comment states `at the very least`. You could beef up the comment here to clear up any confusion but I think a bit of renaming or refactoring could go a long way.

This is probably too much of a refactor at this point but a bitset feels like a great tool for this type of job:

```go
ReadExpiration = 0b0001
WriteExpiration = 0b0010
ReadSession = 0b0100
WriteSession = 0b0100

DualWrite = WriteExpiration | WriteSession

if mode & WriteExpiration == WriteExpiration {
...
}
```

Maybe just rename `isSessionBasedLeasingModeActive` to `sessionBasedLeasingModeIsAtLeast`?